### PR TITLE
Rewrite library to an ES2015 code style

### PR DESCRIPTION
**Pull request by elisaado**

### CODE REVIEW

File: lib/channel.js [hunk 1]
```diff
@@ -1,65 +1,60 @@
...
-
-function decode(data, encoding) {
-  if(!encoding) return data;
-  switch(encoding) {
-    case 'JSON': return JSON.parse(data);
-    default: throw new Error('Unsupported channel encoding: ' + encoding);
-  }
-}
-
+let EventEmitter = require('events');
```
Comment (WeeJeWel) [inline lib/channel.js]: use const

Comment (WeeJeWel) [reply lib/channel.js]: + `'use strict';` on top

File: lib/channel.js [hunk 2]
```diff
@@ -1,65 +1,60 @@
...
+            encode(data, this.encoding)
+        );
+    }
+
+    close() {
+        this.emit('close');
+    }
+}
+
+function encode(data, encoding) {
```
Comment (WeeJeWel) [inline lib/channel.js]: make static method

File: lib/channel.js [hunk 3]
```diff
@@ -1,65 +1,60 @@
...
+        super();
+
+        this.bus = bus;
+        this.sourceId = sourceId;
+        this.destinationId = destinationId;
+        this.namespace = namespace;
+        this.encoding = encoding;
+
+
+        const onmessage = (sourceId, destinationId, namespace, data) => {
```
Comment (WeeJeWel) [inline lib/channel.js]: Move to class method `_onMessage`

Comment (WeeJeWel) [reply lib/channel.js]: `this._onMessage = this._onMessage.bind(this)`

File: lib/client.js [hunk 4]
```diff
@@ -1,137 +1,135 @@
...
+let CastMessage = protocol.CastMessage;
+
+class Client extends EventEmitter {
+    constructor() {
+        super();
+        this.socket = null;
+        this.ps = null;
+    }
+
+    connect(options) {
```
Comment (WeeJeWel) [inline lib/client.js]: add `async` because a promise is returned

Comment (WeeJeWel) [inline lib/client.js]: `connect({ host })`

File: lib/client.js [hunk 5]
```diff
@@ -1,137 +1,135 @@
...
+    }
+
+    close() {
+        debug('closing connection ...');
+        // using socket.destroy here because socket.end caused stalled connection
+        // in case of dongles going brutally down without a chance to FIN/ACK
+        if (this.socket) this.socket.destroy();
+    }
+
+    send(sourceId, destinationId, namespace, data) {
```
Comment (WeeJeWel) [inline lib/client.js]: deconstruct

File: lib/client.js [hunk 6]
```diff
@@ -1,137 +1,135 @@
...
+        const message = {
+            protocol_version: 0, // CASTV2_1_0
+            source_id: sourceId,
+            destination_id: destinationId,
+            namespace: namespace
```
Comment (WeeJeWel) [inline lib/client.js]: `namespace,`

File: lib/packet-stream-wrapper.js [hunk 7]
```diff
@@ -1,44 +1,43 @@
...
+class PacketStreamWrapper extends EventEmitter {
+  constructor(stream) {
+    super();
+
+    this.stream = stream;
+
+    let state = WAITING_HEADER;
+    let packetLength = 0;
+
+    let self = this;
```
Comment (WeeJeWel) [inline lib/packet-stream-wrapper.js]: Use arrow methods

File: lib/proto.js [hunk 8]
```diff
@@ -1,25 +1,24 @@
...
-  module.exports[message] = {
-    serialize: function(data) {
-      var msg = new extensions[message](data);
-      return msg.encode().toBuffer();
-    },
-    parse: function(data) {
-      return extensions[message].decode(data);
-    }
-  };
+const ProtoBuf  = require("protobufjs");
```
Comment (WeeJeWel) [inline lib/proto.js]: single quotes

File: lib/proto.js [hunk 9]
```diff
@@ -1,25 +1,24 @@
...
+
+let builder = ProtoBuf.loadProtoFile(__dirname + "/cast_channel.proto");
```
Comment (WeeJeWel) [inline lib/proto.js]: Use `path.join(__dirname, 'cast_channel.proto')`

File: lib/proto.js [hunk 10]
```diff
@@ -1,25 +1,24 @@
...
+  'AuthChallenge',
+  'AuthResponse',
+  'AuthError',
+  'DeviceAuthMessage'
+];
+
+messages.forEach((message) => {
+  module.exports[message] = {
+    serialize: (data) => {
+      let msg = new extensions[message](data);
```
Comment (WeeJeWel) [inline lib/proto.js]: lowercase class

Comment (WeeJeWel) [reply lib/proto.js]:
```
const Extension = extensions[message]
const msg = new Extension(data)
```

File: lib/client.js [hunk 11]
```diff
@@ -1,137 +1,135 @@
...
+                this.socket = null;
+                this.ps && this.ps.removeListener('packet', onpacket);
+                this.ps = null;
+                this.emit('close');
+            };
+            this.socket.once('close', onclose);
+        });
+    }
+
+    close() {
```
Comment (WeeJeWel) [inline lib/client.js]: rename to `disconnect()`

**Review by WeeJeWel — CHANGES_REQUESTED** [review]

**Comment by elisaado** [comment]
@WeeJeWel Could you review my latest changes?

**Review by WeeJeWel — APPROVED** [review]

**Comment by elisaado** [comment]
I do not have write access to this repository :sweat_smile:

**Comment by WeeJeWel** [comment]
Now you do!

**Pull request merged**